src/modal/modal.component.ts

```diff
@@ -156,11 +163,23 @@ export class ModalComponent implements OnInit, OnDestroy {
 	constructor(public modalService: ModalService) {}
 
 	/**
-	 * Set document focus to be on the modal component when it is initialized.
+	 * Set modalState on the modal component when it is initialized.
 	 * @memberof ModalComponent
 	 */
 	ngOnInit() {
 		this.modalState = "in";
+	}
+
+	/**
+	 * Set document focus to be on the modal component after it is initialized.
+	 * @memberof ModalComponent
+	 */
+	ngAfterViewInit() {
+		const primaryFocusElement  = this.modal.nativeElement.querySelector(this.selectorPrimaryFocus);
+		if (primaryFocusElement && primaryFocusElement.focus) {
+			primaryFocusElement.focus();
+			return;
+		}
 		this.modal.nativeElement.focus();
```

Comment:
While we're on this topic, when modal is opened, focus goes to first focusable element for accessibility. That code would go here instead focusing the modal element.